.github/workflows/release.yml

```diff
@@ -1,96 +1,141 @@
 ---
-name: armory-release
+# yamllint disable rule:comments-indentation
+# yamllint disable rule:indentation
+
+
+name: 🎁 Armory Release Workflow
 
 on:
   repository_dispatch:
     types: build-and-release
 
+  workflow_dispatch:
+    inputs:
+      dry_run:
+        type: boolean
+        description: Execute a dry-run(no uploads) of the workflow
+        required: true
+        default: true
+    #   release-version:
+    #     description: 'Release Version'
+    #     required: false
+    #     default: '0.0.0'
+
+env:
+  RELEASE_DRY_RUN: "${{ github.event_name != 'repository_dispatch' && github.event.inputs.dry_run }}"
+
 jobs:
   release-wheel:
-    name: Build release wheel
-    runs-on: ubuntu-18.04
+    name: 🛞 Build release wheel
+    runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - name: 🐍 Setup Python 3.9
+        uses: actions/setup-python@v4
         with:
-          fetch-depth: 0
-          ref: ${{ github.event.client_payload.branch }}
-      - uses: actions/setup-python@v1
+          python-version: 3.9
+
+      - name: 🐄 Got git?
+        uses: actions/checkout@v3
         with:
-          python-version: '3.7'
-      - name: Build and release wheel
+          fetch-depth: 0
+
+      - name: 📩 Checkout the Source
+        run: |
+          git fetch --prune --unshallow || true
+          git fetch --depth=1 origin +refs/tags/*:refs/tags/*
+          # Verify .git folder
+          if [ ! -d "./.git" ]; then
+            echo "Expected ./.git folder to exist"
+            exit 1
+          fi
+
+      - name: 🔨 Build wheel
         run: |
-          pip install wheel
-          python setup.py bdist_wheel
-      - uses: pypa/gh-action-pypi-publish@master
+          pip install pip==22.2.2
```

Comment:
pip is moving way to fast this year to pin to a dot release, better would be `>=`